smserver/smserver.go

```diff
@@ -421,3 +432,102 @@ func cloudprotocolLogToPb(log cloudprotocol.PushLog) (pbLog *pb.LogData) {
 
 	return pbLog
 }
+
+// nolint:funlen
+func cloudprotocolAlertToPB(alert *cloudprotocol.AlertItem) (pbAlert *pb.Alert, err error) {
+	pbAlert = &pb.Alert{Tag: alert.Tag, Timestamp: timestamppb.New(alert.Timestamp)}
+
+	switch alert.Tag {
+	case cloudprotocol.AlertTagSystemError:
+		sysAlert, ok := alert.Payload.(cloudprotocol.SystemAlert)
```

Comment:
create separate functions for each alert tag

Reply:
done